### PR TITLE
modules: created a jokes module, fixes #80

### LINKED ISSUE #80: intents/modules: add jokes
Create a joke module or intent.
I'm not sure yet if this should be a module which will retrieve jokes from an API (not necessary I think) or simply an intent.
Opened for contributors.

### PR DESCRIPTION
**Pull request by cvhariharan**
I have created a `jokes` module to fetch a random joke from [https://official-joke-api.appspot.com/random_joke](https://official-joke-api.appspot.com/random_joke).

This can be used by asking something line `tell me a joke` or `make me laugh`.

Currently the url is set as a constant in the module (`jokes.go`) itself. Probably adding it as an env would be the best idea but I couldn't find a way with the given project structure. I need advice here.

### CODE REVIEW

File: modules/jokes.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+var JokesTag = "jokes"
+
+type Joke struct {
+	ID        int64  `json:"id"`
+	Type      string `json:"type"`
+	Setup     string `json:"setup"`
+	Punchline string `json:"punchline"`
+}
+
+// CurrencyReplacer replaces the pattern contained inside the response by the currency of the country
```
Comment (hugolgst) [inline modules/jokes.go:24]: Could you update the comment to match the function's name?

File: modules/jokes.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/olivia-ai/olivia/util"
+)
+
+const JOKEURL = "https://official-joke-api.appspot.com/random_joke"
+
+// JokesTag is the intent tag for its module
+var JokesTag = "jokes"
+
+type Joke struct {
```
Comment (hugolgst) [inline modules/jokes.go:17]: Please comment on the structure

File: modules/jokes.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/olivia-ai/olivia/util"
+)
+
+const JOKEURL = "https://official-joke-api.appspot.com/random_joke"
```
Comment (hugolgst) [inline modules/jokes.go:12]: Since the URL is unexported, you should name it `jokeURL`

**Comment by hugolgst** [comment]
Thanks for this PR! Merged.

**Comment by cvhariharan** [comment]
> Thanks for this PR! Merged.

No problems, this is an amazing project, will contribute more often.

**Comment by hugolgst** [comment]
Thanks a lot! I appreciate that contributors are always welcome!

**Comment by hugolgst** [comment]
Feel free to open issues if you have suggestions or other :)

**Pull request merged**